### ISSUE TITLE
README.md

### ISSUE THREAD

**Issue by duaakhan26** [duaakhan26/Information-Retrieval-Assignemnt-3#1]
<html>
<body>

<hr>
<h2>Yelp Review Star Prediction — Assignment 3</h2>
<h3>Course Project: CS Assignment 3</h3>
<p><strong>Author:</strong> <em>Duaa Khan 2021143</em><br>
<strong>Dataset:</strong> Yelp Academic Dataset<br>
<strong>Models Used:</strong> Multilayer Perceptron (MLP), BERT<br>
<strong>Evaluation Metric:</strong> Accuracy, F1-score, Confusion Matrix</p>
<hr>
<h3>Project Description</h3>
<p>The objective of this project is to evaluate two different machine learning models — <strong>MLP</strong> and <strong>BERT</strong> — on the task of <strong>review star rating prediction</strong> using the Yelp Academic Dataset. This assignment is based on the methodologies surveyed in the provided research paper.</p>
<p>The core task is to predict the <strong>star rating (1 to 5)</strong> of a review based on:</p>
<ul>
<li>
<p>Review <strong>text</strong></p>
</li>
<li>
<p>User and business <strong>metadata</strong> (e.g., review count, average rating, etc.)</p>
</li>
</ul>
<p>Due to size constraints in Colab, I sampled a subset of the dataset and processed it efficiently using TF-IDF and tokenization techniques.</p>
<hr>
<h3>Models</h3>

Model | Description
-- | --
Multilayer Perceptron (MLP) | Uses TF-IDF + business/user metadata as input to a simple feedforward neural network.
BERT (base) | Pretrained transformer model fine-tuned on review text for multiclass classification (5 classes).

<hr>
<h3>How to Run</h3>
<ol>
<li>
<p>Upload the Yelp <code inline="">.json</code> files to a folder in Google Drive.</p>
</li>
<li>
<p>Mount Drive in Colab and follow the notebook steps.</p>
</li>
<li>
<p>Train and evaluate both models using the provided code.</p>
</li>
<li>
<p>Optional: visualize confusion matrices and export results.</p>
</li>
</ol>
<hr>
<h3>✅ Conclusion</h3>
<ul>
<li>
<p><strong>BERT significantly outperforms MLP</strong>, especially in understanding subtle differences between review sentiments.</p>
</li>
<li>
<p>MLP is limited by bag-of-words features and performs poorly on mid-range star ratings.</p>
</li>
<li>
<p>Future work could include trying hybrid models like DeepCoNN or C-DSSM, or incorporating review tips/check-ins.</p>
</li>
</ul>
<hr>
</body>
</html>
